File: source-connector/src/main/java/sourceconnector/config/TopicConfig.java
package sourceconnector.config;

import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties("target.kafka")
public record TopicConfig(
  String offsetTopic,
  String sinkTopic
) {
  public TopicConfig {
    if (offsetTopic == null || offsetTopic.isBlank()) {
      throw new IllegalArgumentException("offsetTopic must not be null or blank");
    }
    if (sinkTopic == null || sinkTopic.isBlank()) {
      throw new IllegalArgumentException("sinkTopic must not be null or blank");
    }
  }
}

File: source-connector/src/main/java/sourceconnector/config/TopicConfig.kt
package sourceconnector.config

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties("target.kafka")
data class TopicConfig(
  val offsetTopic: String,
  val sinkTopic: String
) {
  init {
    require(offsetTopic.isNotBlank()) { "offsetTopic must not be null or blank" }
    require(sinkTopic.isNotBlank()) { "sinkTopic must not be null or blank" }
  }
}
